tests/ui/unsized/issue-115203.rs

```diff
@@ -0,0 +1,11 @@
+// compile-flags: --emit link
```

Comment:
I am unsure about how to label this test header. Only ﻿`run-fail` can reproduce the ice of #115203, but it also requires the case to build successfully. Consequently, I have resorted to using `﻿--emit link`.